### ISSUE TITLE
Revert "Add shell completion for minikube kubectl subcommand"

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#16134]
This PR https://github.com/kubernetes/minikube/pull/15856 was nice but it unfortuantely added 50mb to minikube binary
we need to revert it and find a better way to do it

maybe we could "kubectl completion" somehow instead of importing it

CC: 
@ben-krieger 

### Attach the log file

Slack Chat conversation
https://kubernetes.slack.com/archives/C1F5CT6Q1/p1679394095940159
```

Something got "bazeled" in latest minikube (master), now starlark is adding like 50M to the minikube binary ?
https://github.com/kubernetes/minikube/commit/1bd9b39d5189589541c170a6119434c19ba2c6fd (edited) 
GitHubGitHub
[Add shell completion for minikube kubectl subcommand · kubernetes/minikube@1bd9b39](https://github.com/kubernetes/minikube/commit/1bd9b39d5189589541c170a6119434c19ba2c6fd) (112 kB)
```

### Operating System

None

### Driver

None

**Comment by ben-krieger** [comment]
@medyagh Huh... guess there isn't a way to import kubectl without that. 

Anyway, I'm more than fine with reverting this. There has to be a better way to get completions that include flags, but I haven't found it. Once this is reverted, I'll take a fresh look.

**Comment by medyagh** [comment]
> @medyagh Huh... guess there isn't a way to import kubectl without that.
> 
> Anyway, I'm more than fine with reverting this. There has to be a better way to get completions that include flags, but I haven't found it. Once this is reverted, I'll take a fresh look.

no worries, we are going for a release next week, we could re-do this PR after the release and find a way to be more optimized in terms of binary size. 

@afbjorklund suggested we could just call kubectl binary itself for the completion or if there is a way we could import a subset of the library instead of all of it

**Comment by medyagh** [comment]
@ben-krieger do you think, you could assist in reverting the PR (the auto-revert doesnt work due to a conflict)

**Comment by ben-krieger** [comment]
> > @medyagh Huh... guess there isn't a way to import kubectl without that.
> > 
> > Anyway, I'm more than fine with reverting this. There has to be a better way to get completions that include flags, but I haven't found it. Once this is reverted, I'll take a fresh look.
> 
> no worries, we are going for a release next week, we could re-do this PR after the release and find a way to be more optimized in terms of binary size. 
> 
> @afbjorklund suggested we could just call kubectl binary itself for the completion or if there is a way we could import a subset of the library instead of all of it

Usually the Go compiler is very good at removing unused code paths, so I'm guessing it really is used and the only way will be to use the kubectl binary. 

That said, there was some reason I didn't do that. It's been too long for me to remember why, though.

I do remember that Cobra's API makes it impossible to do noun and flag completion via a callback function. Unfortunately, that probably means getting completion right will have to occur outside of Go. I'll look at what kubectl does exactly.

**Comment by afbjorklund** [comment]
Revert merged, 7f1383fb891561c3562dcbc0949fb5ea6a0e28dc
